scalapact-http4s-0-21/src/main/scala/com/itv/scalapact/http4s21/impl/HttpInstances.scala

```diff
@@ -12,7 +16,24 @@ trait HttpInstances {
   implicit def serverInstance: IPactStubber =
     new PactStubber
 
-  implicit def httpClientBuilder(implicit sslContextMap: SslContextMap): IScalaPactHttpClientBuilder =
+  private val clients: ConcurrentHashMap[ClientConfig, IScalaPactHttpClient] = new ConcurrentHashMap
+
+  implicit def httpClientBuilder(implicit sslContextMap: SslContextMap): IScalaPactHttpClientBuilder = {
     (clientTimeout: Duration, sslContextName: Option[String], maxTotalConnections: Int) =>
-      ScalaPactHttpClient(clientTimeout, sslContextName, maxTotalConnections)
+      val clientConfig = ClientConfig(clientTimeout, sslContextName, maxTotalConnections, sslContextMap)
+      PactLogger.message(s"Checking client cache for config $clientConfig, cache size is ${clients.size}")
```

Comment:
maybe make this debug so it can be turned down

Reply:
Done although I don't think anything other than `QuietPactLogger` is ever used 🤔

Reply:
sure, but the user can override the logging behaviour, by putting `PactLogger.noisyLogger() : Unit` in their tests